src/clue.js

```diff
@@ -2,25 +2,156 @@
+        color: "purple"
+    },
+    {
+        firstName: "Kasandra",
+        lastName: "Scarlet",
+        occupation: "Actor",
+        age: 31,
+        description: "She is an A-list movie star with a dark past",
+        image: "https://www.radiotimes.com/uploads/images/Original/111967.jpg",
+        color: "red"
+    },
+    {
+        firstName: "Eleanor",
+        lastName: "Peacock",
+        occupation: "Socialité",
+        age: 36,
+        description: "She is from a wealthy family and uses her status and money to earn popularity",
+        image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+        color: "blue"
+    },
+    {
+        firstName: "Jack",
+        lastName: "Mustard",
+        occupation: "Retired Football player",
+        age: 62,
+        description: "He is a former football player who tries to get by on his former glory",
+        image: "https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg",
+        color: "yellow"    
+    }
+];
 
 // Rooms Array
 
-const roomsArray = [];
+const roomsArray = [
+    {name: "Dining Room"},
+    {name: "Conservatory"},
+    {name: "Kitchen"},
+    {name: "Study"},
+    {name: "Library"},
+    {name: "Billiard Room"},
+    {name: "Lounge"},
+    {name: "Ballroom"},
+    {name: "Hall"},
+    {name: "Spa"},
+    {name: "Living Room"},
+    {name: "Observatory"},
+    {name: "Theater"},
+    {name: "Guest House"},
+    {name: "Patio"}    
+];
 
 // Weapons Array
 
-const weaponsArray = [];
+const weaponsArray = [
+    {
+        name: "rope",
+        weight: 10
+    },
+    {
+        name: "knife", 
+        weight: 8    
+    },
+    {
+        name: "candlestick",  
+        weight: 2
+    },
+    {
+        name: "dumbbell",  
+        weight: 30
+    },
+    {
+        name: "poison", 
+        weight: 2
+    },
+    {
+        name: "axe", 
+        weight: 15
+    },
+    {
+        name: "bat", 
+        weight: 1
+    },
+    {
+        name: "trophy", 
+        weight: 25
+    },
+    {
+        name: "pistol", 
+        weight: 20
+    }
+];
 
 
 // ITERATION 2
 
-function selectRandom() {}
+function selectRandom(arr) {
+    if(arr.length===0){
+        return undefined
+    }
+    let randomElement = arr[Math.floor(Math.random()*arr.length)]
+    return randomElement
+}
 
-function pickMystery() {}
+function pickMystery() {
+    let mystery= {suspect:"",weapon:"",room:""}
+    if(selectRandom(suspectsArray)){
+        mystery.suspect=selectRandom(suspectsArray)
+   
+    }if(selectRandom(weaponsArray)){
+        mystery.weapon=selectRandom(weaponsArray)
+
+    }if(selectRandom(roomsArray)){
+        mystery.room=selectRandom(roomsArray)
+    
+} return mystery
+}
 
 
 // ITERATION 3
 
-function revealMystery() {}
+function revealMystery(mystery) {
+    return `${mystery.suspect.firstName} ${mystery.suspect.lastName} killed Mr. Boddy using the ${mystery.weapon.name} in the ${mystery.room.name}!`
```

Comment:
In this part we can create variables in order to make it easier to read will be somehitng like
```suggestion
const firstName=envelope.suspect.firstName
const lastName=envelope.suspect.lastName
const weapon = envelope.weapon.name
const room = envelope.room.name

  ${firstName} ${.lastName} killed Mr. Boddy using the ${weapon} in the ${room}!!`
```